runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -350,7 +350,7 @@ public void close() throws Exception {
   protected final long getPushbackWatermarkHold() {
     // if we don't have side inputs we never hold the watermark
     if (sideInputs.isEmpty()) {
-      return BoundedWindow.TIMESTAMP_MAX_VALUE.getMillis();
+      return Long.MAX_VALUE;
```

Comment:
Since this is Flink-only code I figured it might be OK. But just quoting the javadoc, the choice of a maximum is so you can safely represent it even if your runner uses higher precision.